crates/pixi_manifest/src/manifests/table_name.rs

```diff
@@ -65,29 +66,34 @@ impl TableName<'_> {
         let mut parts = Vec::new();
 
         if self.prefix.is_some() {
-            parts.push(self.prefix.unwrap());
+            parts.push(self.prefix.unwrap().to_string());
         }
 
         if self
             .feature_name
             .as_ref()
             .is_some_and(|feature_name| !feature_name.is_default())
         {
-            parts.push("feature");
-            parts.push(
-                self.feature_name
-                    .as_ref()
-                    .expect("we already verified")
-                    .as_str(),
-            );
+            parts.push("feature".to_string());
+            let feature_str = self
+                .feature_name
+                .as_ref()
+                .expect("we already verified")
+                .as_str();
+
+            // Use the utility function to escape the feature name if needed
+            parts.push(escape_toml_key(feature_str));
```

Comment:
Instead of heap allocating all other strings. You can assign the result of this function to an uninitialized let binding and then take the reference of that binding to store in the result vec.

Reply:
You can create a non mut let binding _without initializing it_. Then in the body of the if, assign it.